.github/workflows/publish-to-npm.yaml

```diff
@@ -0,0 +1,36 @@
+name: Publish release to npm
+
+on:
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Publish to npm
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.2.2
+
+      - name: Setup Node
+        uses: actions/setup-node@v4.1.0
+        with:
+          node-version-file: .nvmrc
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Generate tag
+        id: generate_tag
+        run: |
+          NPM_TAG=$(npm run generate-npm-tag --silent)
```

Comment:
The existing generate-npm-tag script maybe isn't as robust as we'd like: It checks the highest published git tag, gets the current package.json version, then sets the tag to `latest` if the package.json version is higher, or `latest-[major-number]` if it's lower.